extensions/spring-cloud-config-client/runtime/src/main/java/io/quarkus/spring/cloud/config/client/runtime/VertxSpringCloudConfigGateway.java

```diff
@@ -0,0 +1,257 @@
+    }
+
+    private URI determineBaseUri(SpringCloudConfigClientConfig springCloudConfigClientConfig) throws URISyntaxException {
+        String url = springCloudConfigClientConfig.url;
+        if (null == url || url.isEmpty()) {
+            throw new IllegalArgumentException(
+                    "The 'quarkus.spring-cloud-config.url' property cannot be empty");
+        }
+        if (url.endsWith("/")) {
+            return new URI(url.substring(0, url.length() - 1));
+        }
+        return new URI(url);
+    }
+
+    private URI finalURI(String applicationName, String profile) throws URISyntaxException {
+        URIBuilder result = new URIBuilder(baseURI);
+        if (result.getPort() == -1) {
+            // we need to set the port otherwise auth case doesn't match the request
+            result.setPort(result.getScheme().equalsIgnoreCase("http") ? 80 : 443);
+        }
+        List<String> finalPathSegments = new ArrayList<>(result.getPathSegments());
+        finalPathSegments.add(applicationName);
+        finalPathSegments.add(profile);
+        if (springCloudConfigClientConfig.label.isPresent()) {
+            finalPathSegments.add(springCloudConfigClientConfig.label.get());
+        }
+        result.setPathSegments(finalPathSegments);
+        return result.build();
+    }
+
+    @Override
+    public Uni<Response> exchange(String applicationName, String profile) throws Exception {
+        final URI finalURI = finalURI(applicationName, profile);
+        String label = "";
+        if (springCloudConfigClientConfig.label.isPresent()) {
+            label = springCloudConfigClientConfig.label.get();
+        }
+        HttpRequest<Buffer> request = webClient
+                .get(finalURI.getPort(), finalURI.getHost(), "/" + applicationName + "/" + profile + "/" + label)
+                .ssl(true)
+                .putHeader("Accept", "application/json;charset=UTF-8");
+        for (Map.Entry<String, String> entry : springCloudConfigClientConfig.headers.entrySet()) {
+            request.putHeader(entry.getKey(), entry.getValue());
+        }
+        log.debug("Attempting to read configuration from '" + finalURI + "'.");
+        return request.send().map(r -> {
+            if (r.statusCode() != 200) {
+                throw new RuntimeException("Got unexpected HTTP response code " + r.statusCode()
+                        + " from " + finalURI);
+            } else {
+                String bodyAsString = r.bodyAsString();
+                if (bodyAsString.isEmpty()) {
+                    throw new RuntimeException("Got empty HTTP response body " + finalURI);
+                }
+                try {
+                    return OBJECT_MAPPER.readValue(bodyAsString, Response.class);
+                } catch (JsonProcessingException e) {
+                    throw new RuntimeException("Got unexpected error " + e.getOriginalMessage());
+                }
+            }
+        });
+    }
+
+    //    @Override
```

Comment:
Better remove this commented out code completely

Reply:
Oups! thank you :-)